zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -155,62 +157,103 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             code_hash.expr(),
         );
 
-        // Setup first call's context.
-        for (field_tag, value) in [
-            (CallContextFieldTag::Depth, 1.expr()),
-            (CallContextFieldTag::CallerAddress, tx_caller_address.expr()),
-            (CallContextFieldTag::CalleeAddress, tx_callee_address.expr()),
-            (CallContextFieldTag::CallDataOffset, 0.expr()),
-            (
-                CallContextFieldTag::CallDataLength,
-                tx_call_data_length.expr(),
+        let is_empty_code_hash = IsEqualGadget::construct(
+            cb,
+            code_hash.expr(),
+            Word::random_linear_combine_expr(
+                (*EMPTY_HASH_LE).map(|byte| byte.expr()),
+                cb.power_of_randomness(),
             ),
-            (CallContextFieldTag::Value, tx_value.expr()),
-            (CallContextFieldTag::IsStatic, 0.expr()),
-            (CallContextFieldTag::LastCalleeId, 0.expr()),
-            (CallContextFieldTag::LastCalleeReturnDataOffset, 0.expr()),
-            (CallContextFieldTag::LastCalleeReturnDataLength, 0.expr()),
-            (CallContextFieldTag::IsRoot, 1.expr()),
-            (CallContextFieldTag::IsCreate, tx_is_create.expr()),
-            (CallContextFieldTag::CodeHash, code_hash.expr()),
-        ] {
-            cb.call_context_lookup(true.expr(), Some(call_id.expr()), field_tag, value);
-        }
+        );
+
+        cb.condition(is_empty_code_hash.expr(), |cb| {
+            cb.require_equal(
+                "Tx to account with empty code should be successful",
```

Comment:
This string would make me think that the check is for `is_success`.  I know that in this context `is_persistent == is_success`.  What do you think about saying "should be persistent" instead?

Reply:
Make sense, fixed in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/872/commits/ba855da94f8ee361e82456f880de5a1beb7a1f18.